mlflow/recipes/cards/__init__.py

```diff
@@ -237,11 +237,19 @@ def render_table(table, columns=None, hide_index=True):
             in the output HTML table.
         :param hide_index: Hide index column when rendering.
         """
+        import html
+
         import pandas as pd
         from pandas.io.formats.style import Styler
 
         if not isinstance(table, Styler):
-            table = pd.DataFrame(table, columns=columns).style
+            table = pd.DataFrame(table, columns=columns)
+            # Escape specific characters in HTML to prevent
+            # javascript code injection
+            # Note that `pandas_df.style.to_html(escape=True) does not work
```

Comment:
why `pandas_df.style.to_html(escape=True)` does not work? What's the issue?

Reply:
I think it is a pandas library bug. We can report it.

Reply:
what happens if we use `pandas_df.style.to_html(escape=True)`?

Reply:
It generate the same result with `pandas_df.style.to_html(escape=False)`

But `pandas_df.to_html(escape=True)` generate escaped result correctly.

So I think it is a bug.

You can try it by:

```
p1 = pd.DataFrame({'a': ['<div><script></script></div>']})

print(p1.style.to_html(escape=False))
print(p1.style.to_html(escape=True))

print(p1.to_html(escape=False))
print(p1.to_html(escape=True))
```